MOBILE-3320: Navigation tests

NoelDeMartin opened the pull request:
Before this is merged https://github.com/moodlehq/moodleapp/pull/2823, should be resolved.

dpalou commented on tests/behat/behat_app.php:
```diff
@@ -486,10 +505,10 @@ public function i_press_the_standard_button_in_the_app(string $button) {
      * @param TableNode $data
      */
-    public function i_receive_a_forum_push_notification(TableNode $data) {
+    public function i_receive_a_push_notification(TableNode $data) {
```
Why are you removing "forum" from the name? If in the future we want to test other push notifications like messages we'll have to rename the function again. IMO it'd be better if this function and i_open_a_custom_link specify that they're for forum.

NoelDeMartin replied:
Actually I noticed I didn't remove it from the step definition, just the function name, but I intended to remove it from both places.

I did it because looking at the implementation, we can actually use this method for all types of push notifications by checking the type of the activity (for example, the "discussion" column could be optional in the input table).

I guess we could potentially do the same in the custom link step, but I intended to leave that one specific to forums (for now) because it's only taking a discussion title as an argument.

I'll take a look next week and use the same strategy for both, we can leave this PR open until then.

NoelDeMartin replied:
I have standarized this, I decided to use table format so that the same step can be reused in the future without breaking existing tests or needing to add a lot of step definitions. The table syntax is certainly less readable, but for this types of tests that are not used so often I think the flexibility is a good trade-off for readability. One example of this is the database setup tests, that use table syntax.

NoelDeMartin commented:
@dpalou Other than fixing the issue you commented on your review, I've also added a couple of new commits related with some new functionality I've added in this PR: https://github.com/moodlehq/moodleapp/pull/2832
